pkg/rc/farm.go

```diff
@@ -108,37 +105,32 @@ func (rcm *Farm) Start(quit <-chan struct{}) {
 			// track which children were found in the returned set
 			foundChildren := make(map[fields.ID]struct{})
 			for _, rcField := range rcFields {
+				rcLogger := rcm.logger.SubLogger(logrus.Fields{
+					"rc_id": rcField.ID,
+				})
 				if _, ok := rcm.children[rcField.ID]; ok {
 					// this one is already ours, skip
-					rcm.logger.WithField("rc_id", rcField.ID).Debugln("Got replication controller already owned by self")
+					rcLogger.NoFields().Debugln("Got replication controller already owned by self")
 					foundChildren[rcField.ID] = struct{}{}
 					continue
 				}
 
-				success, err := rcm.rcStore.LockRead(rcField.ID, session)
-				if err != nil {
-					// log and break out of the loop - we can't continue if
-					// we had a real failure, our session probably disappeared
-					rcm.logger.WithErrorAndFields(err, logrus.Fields{
-						"rc_id":   rcField.ID,
-						"session": session,
-					}).Errorln("Got error while locking replication controller - session may be expired")
-					break // TODO: should we be pruning after this?
-				} else if !success {
+				err := rcm.lock.Lock(kp.LockPath(kp.RCPath(rcField.ID.String())))
+				if _, ok := err.(kp.AlreadyLockedError); ok {
 					// someone else must have gotten it first - log and move to
 					// the next one
-					rcm.logger.WithFields(logrus.Fields{
-						"rc_id":   rcField.ID,
-						"session": session,
-					}).Debugln("Lock on replication controller was denied")
+					rcLogger.NoFields().Debugln("Lock on replication controller was denied")
 					continue
+				} else if err != nil {
```

Comment:
in this case (if it happened on all servers running the rc farm) is there any mechanism by which this orphaned replication controller will get a parent? any retrying anywhere?

I guess WatchNew returns the whole list so a newly created replication controller should trigger a retry?